pkg/storage/cache.go

```diff
@@ -62,35 +61,43 @@ func NewInMemoryLRUCache[T any](opts ...InMemoryLRUCacheOpt[T]) *InMemoryLRUCach
 		opt(t)
 	}
 
-	t.ccache = ccache.New(ccache.Configure[T]().MaxSize(t.maxElements))
+	cache, _ := theine.NewBuilder[string, T](t.maxElements).Build()
+	t.cache = cache
 	return t
 }
 
 func (i InMemoryLRUCache[T]) Get(key string) T {
 	var zero T
-	item := i.ccache.Get(key)
-	if item == nil {
+	if i.cache == nil {
 		return zero
 	}
 
-	if value, expired := item.Value(), item.Expired(); !reflect.ValueOf(value).IsZero() && !expired {
-		return value
+	value, ok := i.cache.Get(key)
+	if !ok {
+		return zero
 	}
-
-	return zero
+	return value
 }
 
 func (i InMemoryLRUCache[T]) Set(key string, value T, ttl time.Duration) {
-	i.ccache.Set(key, value, ttl)
+	if ttl <= 0 {
```

Comment:
In what case do we pass in negative TTL?

Reply:
Negative TTL handling is tested in cache_test.go in the "set_and_get_expired" test case. When a negative TTL is passed, we treat it as a delete operation - this maintains consistent behavior with the previous implementation and provides a clean way to handle invalid TTL values.


https://github.com/Siddhant-K-code/openfga/blob/8c030dc5fff1610c86c47a1de2991559b36c94e0/pkg/storage/cache_test.go#L25-L34